src/api/app/controllers/webui/user_controller.rb

```diff
@@ -22,7 +22,15 @@ def show
     @role_titles = @displayed_user.roles.global.pluck(:title)
     @account_edit_link = CONFIG['proxy_auth_account_page']
 
-    switch_to_webui2
+    return unless switch_to_webui2
+    @last_day = Time.zone.today
+
+    # take the last 52 weeks
+    @first_day = @last_day - (52 * 7)
+    # move back to the monday before (make it up to 53 weeks)
+    @first_day -= (@first_day.cwday - 1)
+
+    @activity_hash = User::Contributions.new(@displayed_user, @first_day).activity_hash
```

Comment:
Why do you calculate this every time? The past never changes.

Reply:
But today does change

Reply:
True